### PR TITLE
Record recognizes motor stop while jogging

### PR DESCRIPTION
**Pull request by tboegi**
The following situation came up while debugging soft limits in a
motion controller which had different values then the record.

The motor is at position 50mm (RBV=50), the low soft limit is 15 (LLM=15),
jogging velocity is 10 mm/sec (JVEL=10).

The motion controller has an internal soft limit which is 35 mm, which
we want to test.
Set JOGR to 1, the motor will move backwards.
(Side note: it would be stopped by the record once it passed 25mm).

Before that, the motion controller stops the motor internally and reports DONE.
The state machine in the record is not prepared to handle this and JOGR stays 1.
The expected behavior is that JOGR returns to 0.
Because this is an unexpected stop, reset even pending home requests.
In other words: call clear_buttons(pmr)

Whether or not an alarm is raised may be another question.
The natural thing would be to set LVIO to 1, but we don't know why the controller
stopped the motion.
Unless we add a bit in MSTA like "RA_LVIO"...
But in any case this can and should go into a separate commit.

### CODE REVIEW

**Comment by rsluiter** [comment]
"The motion controller has an internal soft limit... which we want to test."  Temporarily disabling the motor record's soft limits (by setting DHLM = DLLM = 0) is a straight-forward way to test the motor controller's internal soft limits. Otherwise, the assumption has always been that, if motor controller's soft limits are used, the motor record's soft limits should be within the valid range of the controller's limits. If that convention is followed, there is no need for this change.

**Comment by kmpeters** [comment]
> Otherwise, the assumption has always been that, if motor controller's soft limits are used, the motor record's soft limits should be within the valid range of the controller's limits. If that convention is followed, there is no need for this change.

Is there a downside to setting the MIP to done when a status update detects that the axis is done moving, even though the user is still holding down the jog button?  I'm struggling to find one.

**Comment by rsluiter** [comment]
I also do not see a "downside to setting the MIP to done when a status update detects that the axis is done moving, even though the user is still holding down the jog button", unless there are unintended consequences to this change. I recommend extensive testing before committing this change.

**Comment by kmpeters** [comment]
Without the change in this pull request, I confirm that a motor record that is stopped while jogging stops, the target position is synced with the readback position, and the JOG{R,F} field is left at 1.

With the change in this pull request, a motor record that is stopped while jogging stops, the JOG{R,F} field is reset to 0, but the target position is *not* synced with the readback position.

I'm looking into the the best way to retain the syncing of the target position, which is the standard behavior when a jog is complete.

**Comment by tboegi** [comment]
Thanks for testing -
I will have a look as well.

**Comment by kmpeters** [comment]
Commenting out the the `goto process_exit;` that was added allows the motor record to reset the jog field and sync the target position.  It isn't obvious to me why this is the case, but I haven't run my ioc in the debugger yet.

**Comment by tboegi** [comment]
I send a new patch (with the same branch name) without the goto process_exit;
We need to go through postProcess() to synch VAL from RBV, and the new patch does this.
